src/pages/api/stripe/webhook.ts

```diff
@@ -17,8 +57,90 @@ const stripeWebhookHandler = async (
     try {
       event = stripe.webhooks.constructEvent(buf, sig as string, webhookSecret)
 
-      if (event.type === 'checkout.session.completed') {
-        console.log('stripe webhook handler', {event})
+      // Also handle:
+      // - 'customer.subscription.updated'
+      // - 'customer.subscription.deleted'
+      if (event.type === 'customer.subscription.updated') {
+        const stripeSubscription = await stripe.subscriptions.retrieve(
+          event.data.object.id,
+        )
+        const stripeCustomer = await stripe.customers.retrieve(
+          event.data.object.customer,
+        )
+        const previousSubscription = event.data.previous_attributes
+        const newPlanAmount = stripeSubscription.items.data[0].plan.amount || 0
+
+        const state =
+          previousSubscription.plan.amount < newPlanAmount
+            ? 'upgrade'
+            : 'downgrade'
```

Comment:
felt like this is what we want to know and this should answer it

I tested it a bit and it seems to do the right thing